tests/e2e/declarative_database_management_test.go

```diff
@@ -101,65 +100,147 @@ var _ = Describe("Declarative databases management test", Label(tests.LabelSmoke
 			})
+		})
+		When("Database CRD reclaim policy is set to delete", func() {
+			It("can manage a declarative database and delete it in Postgres", func() {
+				// NOTE: the Postgres database 'declarative' created in the previous spec
+				// was retained after deletion.
+				// This manifest adopts the existing database and only changes the retention policy
+				databaseManifest := fixturesDir +
+					"/declarative_databases/database-with-delete-reclaim-policy.yaml.template"
+				shouldPgDatabaseBeRetained := false
+				expectedDatabaseFields := apiv1.Database{
+					Spec: apiv1.DatabaseSpec{
+						Name:      "declarative",
+						LcCtype:   "en_US.utf8",
+						LcCollate: "C", // this is the default value
+						Encoding:  "0", // corresponds to SQL_ASCII
+					},
+				}
+
+				assertTestDeclarativeDatabase(databaseManifest,
+					shouldPgDatabaseBeRetained, expectedDatabaseFields)
+			})
+		})
+	})
+
+	Context("in a Namespace to be deleted manually", func() {
+		const (
+			namespace = "declarative-db-finalizers"
+		)
+		var (
+			err                error
+			clusterName        string
+			databaseObjectName string
+		)
+		It("will not prevent the deletion of the namespace with lagging finalizers", func() {
+			By("setting up the new namespace and cluster", func() {
+				err = env.CreateNamespace(namespace)
+				Expect(err).ToNot(HaveOccurred())
 
-			It("keeps the db when Database CRD is removed", func() {
-				By("remove Database CRD", func() {
-					Expect(utils.DeleteObject(env, database)).To(Succeed())
-				})
+				clusterName, err = env.GetResourceNameFromYAML(clusterManifest)
+				Expect(err).ToNot(HaveOccurred())
 
-				By("verifying database is still existing", func() {
-					primaryPodInfo, err := env.GetClusterPrimary(namespace, clusterName)
-					Expect(err).ToNot(HaveOccurred())
+				AssertCreateCluster(namespace, clusterName, clusterManifest, env)
+			})
+			By("creating the database", func() {
+				databaseManifest := fixturesDir +
+					"/declarative_databases/database-with-delete-reclaim-policy.yaml.template"
+				databaseObjectName, err = env.GetResourceNameFromYAML(databaseManifest)
+				Expect(err).NotTo(HaveOccurred())
+				CreateResourceFromFile(namespace, databaseManifest)
+			})
+			By("ensuring the database is reconciled successfully", func() {
+				// get database object
+				dbObj := &apiv1.Database{}
+				databaseNamespacedName := types.NamespacedName{
+					Namespace: namespace,
+					Name:      databaseObjectName,
+				}
 
-					assertDatabaseExists(namespace, primaryPodInfo.Name, dbname, true)
-				})
+				Eventually(func(g Gomega) {
+					err := env.Client.Get(env.Ctx, databaseNamespacedName, dbObj)
+					g.Expect(err).ToNot(HaveOccurred())
+					g.Expect(dbObj.Status.Ready).Should(BeTrue())
+				}, 300).WithPolling(10 * time.Second).Should(Succeed())
+			})
+			By("deleting the namespace and making sure it succeeds before timeout", func() {
+				err := env.DeleteNamespaceAndWait(namespace, 30)
```

Comment:
```suggestion
				err := env.DeleteNamespaceAndWait(namespace, 60)
```

Reply:
Yep, I started with 60 seconds, but after exercising the suite I guess it is fine to reduce it.